packages/markdown/remark/src/remark-shiki.ts

```diff
@@ -30,9 +30,15 @@ export interface ShikiConfig {
 	wrap?: boolean | null;
 }
 
-const remarkShiki = async ({ langs = [], theme = 'github-dark', wrap = false }: ShikiConfig) => {
-	const highlighter = await getHighlighter({ theme });
+/** 
+ * getHighlighter() is the most expensive step of Shiki. Instead of calling it on every page,
+ * cache it here as much as possible. Make sure that your highlighters can be cached, state-free.
+ */
+const HIGHLIGHTER_CACHE: Record<string, shiki.Highlighter> = {};
 
+const remarkShiki = async ({ langs = [], theme = 'github-dark', wrap = false }: ShikiConfig) => {
+	const cacheID: string = typeof theme === 'string' ? theme : theme.name;
+  	const highlighter = HIGHLIGHTER_CACHE[cacheID] = HIGHLIGHTER_CACHE[cacheID] || await getHighlighter({ theme });
```

Comment:
I'd prefer to:
1. Use `Map<string, shiki.Highlighter>` for the cache rather than an Object.
2. Use a more readable `if` statement rather than this assignment pattern.

Reply:
done!